### ISSUE TITLE
Do not rebind C-d and C-k

### ISSUE THREAD

**Issue by salutis** [bdarcus/citar#503]
In Emacs, `DEL` deletes character backward and `C-DEL` deletes them forward. Why do we rebind `C-DEL` default to `citar-org-delete-citation`? Does Org bring this (bad) idea or does Citar bring it?

**Comment by bdarcus** [comment]
Propose an alternative?

https://github.com/bdarcus/citar#org-cite-1

**Comment by salutis** [comment]
We add citations with `C-c C-x @`. How about we delete them with `C-c C-x DEL`?

**Comment by bdarcus** [comment]
There's a couple of wrinkles though:

1. Keybindings vary quite a bit, most obviously between vanilla and evil. So if we move more towards one, then we make them less useful for the other.
2. That command, at least, is contextual. I think the bindings are too?
3. They can be locally customized.

Not sure what I conclude from all of the above.

**Comment by salutis** [comment]
I  think we should _not_ hijack the most basic editing key bindings such as "delete character". Today, keyboards often do not have a dedicated "Delete" button, or the user does not know how to use it. (They all still have a backspace button, thanks goodness.) See laptops, small keyboards, etc. Every time I want to delete a single character in a citation, `citar` deletes the whole thing. Poof! I do not think that makes for a good _default_ behavior.

**Comment by salutis** [comment]
BTW, the same applies to `citar-org-kill-citation` that hijacks `C-k`. I have just tried to kill a line (that starts with a citation), and the computer deleted just the citation. These sort of drastic remaps of the most basic keys should activate when the user explicitly enables them, sort of what Org does with `org-return-follows-link` that defaults to `nil`.

**Comment by salutis** [comment]
TL;DR As a user, I feel like I need to pay _super-close_ attention when editing text with Citar, because the most basic keyboard shortcuts behave differently around citations. I feel like I can never be sure what any given keyboard shortcut will do.

**Comment by bdarcus** [comment]
Another option is just remove the keymap, since I really don't think the solution is as straightforward as you suggest (basically prioritize vanilla keybindings).

Any thoughts @roshanshariff @localauthor?

**Comment by bdarcus** [comment]
We could also have two keymaps for this, with the choice configurable?

**Comment by salutis** [comment]
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Old</th>
<th scope="col" class="org-left">New</th>
<th scope="col" class="org-left">Description</th>
</tr>
</thead>

<tbody>
<tr>
<td class="org-left"><code>&lt;mouse-1&gt;</code></td>
<td class="org-left"><code>&lt;mouse-1&gt;</code></td>
<td class="org-left"><code>org-open-at-point</code></td>
</tr>


<tr>
<td class="org-left"><code>&lt;mouse-3&gt;</code></td>
<td class="org-left"><code>&lt;mouse-3&gt;</code></td>
<td class="org-left"><code>embark-act</code></td>
</tr>


<tr>
<td class="org-left"><code>C-d</code></td>
<td class="org-left"><code>C-c C-x DEL</code></td>
<td class="org-left"><code>citar-org-delete-citation</code></td>
</tr>


<tr>
<td class="org-left"><code>C-k</code></td>
<td class="org-left"><code>C-c C-x k</code></td>
<td class="org-left"><code>citar-org-kill-citation</code></td>
</tr>


<tr>
<td class="org-left"><code>M-p</code></td>
<td class="org-left"><code>M-p</code></td>
<td class="org-left"><code>citar-org-update-pre-suffix</code></td>
</tr>


<tr>
<td class="org-left"><code>S-&lt;left&gt;</code></td>
<td class="org-left"><code>S-&lt;left&gt;</code></td>
<td class="org-left"><code>citar-org-shift-reference-left</code></td>
</tr>


<tr>
<td class="org-left"><code>S-&lt;right&gt;</code></td>
<td class="org-left"><code>S-&lt;right&gt;</code></td>
<td class="org-left"><code>citar-org-shift-reference-right</code></td>
</tr>
</tbody>
</table>

(We could use also `C-c C-x d` for delete.  Emacs/Org use both `d` and
`DEL` for deletion. Note that `C-c C-x )

I do not have a keyboard with arrow keys, so I have no idea about
`S-<left>` and `S-<right>`.  That said, I would like all functionality
to work without the arrow keys, like everywhere in Org and Emacs.

As for `M-p`, that stands for "previous" in all contexts I can think of.
Does anyone use it?  It does little and seems broken.  I have just tried
to use `M-p` for the first time.  For `[cite:@bennett+2015]`, I added
"x" suffix, and it changed my citation into `[cite:@bennett+2015x;]`,
rendering it broken.

**Comment by roshanshariff** [comment]
For what it's worth, I agree with @salutis that the keymap should not rebind basic editing commands. I think the suggestions above are reasonable enough? It seems conventional for Org commands to bind themselves under the `C-c C-x` prefix.

**Comment by bdarcus** [comment]
OK, my tentative plan is to move those two controversial bindings to the embark at point map, and add a note to the README about this.

**Comment by roshanshariff** [comment]
Those commands are org-specific, so moving them to the Embark keymap will require new mode-independent commands. Instead, you could just use the standard `delete-region` and `kill-new` commands which I believe do the right thing? (They're also bound in `embark-general-map`)

**Comment by bdarcus** [comment]
> Those commands are org-specific ...

Not the kill/copy-reference one?

**Comment by roshanshariff** [comment]
Aren't they `citar-org-kill-citation` and `citar-org-delete-citation`? They only work in Org mode, right?

**Comment by bdarcus** [comment]
Oh yes; that's right.

**Comment by salutis** [comment]
For a practical example, consider the problem I had a couple of minutes ago. 

Note: `|` stands for point.

I start with the following:

```
* TODO Study [cite:@xxx+2011 210-220]| (the reader's perspective)
```

I press `C-k` to delete ` (the reader's perspective)` and get:

```
* TODO Study |(the reader's perspective)
```

... instead of ...

```
* TODO Study [cite:@xxx+2011 210-220]|
```

**Comment by bdarcus** [comment]
I understand, but you have always been able to write your own keymap for this; still can.

**Comment by salutis** [comment]
Ping!  Almost every day, I delete the entire citation with `C-d` instead of a single character.  Please, consider not overriding such a basic keyboard shortcut.  (I know I can remap, but I speak for all the others who use `C-d`, not just me.)

**Comment by bdarcus** [comment]
I've not forgot; just been busy.

And yeah, you can just do this in your config ;-)

**Comment by bdarcus** [comment]
> I understand, but you have always been able to write your own keymap for this; still can.
> 
> EDIT: or maybe not. If not, I'll definitely fix that.

Just taking a moment to think about this again, perhaps this is a better path as a first step?

**Comment by salutis** [comment]
@bdarcus Thank you so much!

**Comment by RagnarGrootKoerkamp** [comment]
The readme also needs to be updated for this I think.

**Comment by bdarcus** [comment]
> The readme also needs to be updated for this I think.

Correct; I noticed that yesterday.
